docs/how-to/using-groups.rst

```diff
@@ -0,0 +1,231 @@
+dependency solver treat them as a single block. This is useful for
+jobs that require a setup/teardown, or a sequence of jobs that
+must not be interleaved with others.
+
+Key behaviors of groups
+--------------------------
+
+- Jobs in the same group are always run as a contiguous block.
+- Dependencies between jobs **inside** the group are resolved normally.
+- If a job **inside** the group depends on a job **outside**, the **whole group**
+  depends on that outside job.
+- If a job **outside** the group depends on a job **inside**, it depends on the
+  **whole group**.
+- If these group-level dependencies create a cycle, Checkbox raises a
+  dependency error and removes the involved jobs/groups from the plan.
+
+
+Examples
+--------
+
+Setup/teardown structure
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+The group field could be used to create a setup → tests → teardown structure,
+where the setup and teardown jobs are part of the same group as the tests.
+
+.. code-block::
+
+    id: wireless_setup
+    flags: simple
+    group: wireless_tests
+    command: echo 'Setting up wireless tests' 
+
+    id: wireless_test_1
+    flags: simple
+    group: wireless_tests
+    after: wireless_setup
+    before: wireless_teardown
+    command: echo 'Running wireless test 1'
+
+    id: wireless_test_2
+    flags: simple
+    group: wireless_tests
+    after: wireless_setup
+    before: wireless_teardown
+    command: echo 'Running wireless test 2'
+
+    id: wireless_teardown
+    flags: simple
+    group: wireless_tests
+    after: wireless_setup
+    command: echo 'Tearing down wireless tests'
+
+    id: wireless_test_plan
+    name: wireless_test_plan
+    unit: test plan
+    _summary: Setup/teardown group test
+    include:
+      wireless_.*
+
+Execution order::
+
+    wireless_setup
+    wireless_test_1
+    wireless_test_2
+    wireless_teardown
+
+Step by step execution
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If we want to ensure a strict step-by-step execution order between jobs, even
+if there are dependencies on other jobs, we can use groups to enforce that.
+
+.. code-block::
+
+    id: gather_device_info
+    flags: simple
+    command: echo "device_info" >> "$PLAINBOX_SESSION_SHARE"/device_info.txt
+
+    id: device_insert
+    flags: simple
+    group: device_test
+    command: echo "Inserting device"
+
+    id: device_test_write 
+    flags: simple
+    group: device_test
+    depends: gather_device_info device_insert
+    command: echo "Testing device write"
+
+    id: device_test_read
+    flags: simple
+    group: device_test
+    after: device_test_write
+    command: echo "Testing device read"
+
+    id: device_remove
+    flags: simple
+    group: device_test
+    after: device_test_read
+    command: echo "Removing device"
+
+    id: device_test_plan
+    name: device_test_plan
+    unit: test plan
+    _summary: Step by step device test
+    include:
+      device_.*
+      gather_device_info
```

Comment:
This is interesting, because although you put `gather_device_info` after the `device_*` jobs, it ends up being executed before the whole group. It would be good to add a little paragraph to explain this, something like

> Although `gather_device_info` is placed after the `device_*` jobs, it will be executed by Checkbox before them because `device_test_write` depends on it and because it's part of the `device_test` group like all the other `device_*` jobs.